### PR TITLE
overlay: coreos-autologin-generator: add /etc/issue.d for interactive live env

### PR DESCRIPTION
**Pull request by dustymabe**
This issue.d will give the user some information about what is possible
to do in the interactive live environment, including some information
about installing to disk.

### CODE REVIEW

**Comment by dustymabe** [comment]
See also the first commit (` overlay: coreos-autologin-generator: update console log notes`) where I update/clarify the comments around changing the kernel console debug logging.

File: overlay.d/05core/usr/lib/systemd/system-generators/coreos-autologin-generator [hunk 1]
```diff
@@ -32,12 +32,11 @@ ExecStart=-/sbin/agetty --autologin core -o '-p -f core' ${args} %I \$TERM
 EOF
 }
 
-# We can remove this when we fix https://github.com/coreos/fedora-coreos-tracker/issues/220
-silence_audit_on_console() {
-    mkdir -p /run/sysctl.d
```
Comment (jlebon) [inline overlay.d/05core/usr/lib/systemd/system-generators/coreos-autologin-generator:37]: Looks like we lost this `mkdir` here?

**Comment by dustymabe** [comment]
Preview:

coreos-installer

File: overlay.d/05core/usr/lib/systemd/system-generators/coreos-autologin-generator [hunk 2]
```diff
@@ -43,6 +43,30 @@ kernel.printk=4
...
+and installing persistently to disk. Here is an example of running an
+install to disk via `coreos-installer`:
+
+curl -O https://example.com/example.ign
+sudo coreos-installer install /dev/sda \
+         --ignition-file ./example.ign \
+         --image-url https://example.com/image.raw.xz
+
+You may configure networking via `sudo nmcli` or `sudo nmtui` and have
+that configuration persist into an installed system by passing the
```
Comment (jlebon) [inline overlay.d/05core/usr/lib/systemd/system-generators/coreos-autologin-generator]: Minor: s/an/the/ ?

**Comment by dustymabe** [comment]
hmm. from the screenshot it seems like the line continuation backslashes (`\`)aren't showing up. Will investigate.

File: overlay.d/05core/usr/lib/systemd/system-generators/coreos-autologin-generator [hunk 3]
```diff
@@ -43,6 +43,30 @@ kernel.printk=4
...
+`--copy-network` argument to `coreos-installer`. Please run
```
Comment (jlebon) [inline overlay.d/05core/usr/lib/systemd/system-generators/coreos-autologin-generator]: Minor: `coreos-installer install`

File: overlay.d/05core/usr/lib/systemd/system-generators/coreos-autologin-generator [hunk 4]
```diff
@@ -43,6 +43,30 @@ kernel.printk=4
...
+`coreos-installer install --help` for more information on the possible
+options to `coreos-installer`.
```
Comment (jlebon) [inline overlay.d/05core/usr/lib/systemd/system-generators/coreos-autologin-generator]: Maybe just "on the possible install options"? So we avoid repeating `coreos-installer` again.

**Review by jlebon — APPROVED** [review]
This is nice! I like the idea of having a friendly welcome message in the live environment.

LGTM overall, just some minor nits.

**Comment by dustymabe** [comment]
fixed

File: overlay.d/05core/usr/lib/systemd/system-generators/coreos-autologin-generator (hunk 1 above)
Comment (dustymabe) [reply overlay.d/05core/usr/lib/systemd/system-generators/coreos-autologin-generator:37]: yeah - we should have cleaned it up in the other PR when we switched to `/etc/` I can put it in a separate commit if that would be better.

**Comment by imcleod** [comment]
LGreatTM - The direct reference to install process in particular is a welcome addition. Cheers Dusty.

File: overlay.d/05core/usr/lib/systemd/system-generators/coreos-autologin-generator (hunk 2 above)
Comment (dustymabe) [reply overlay.d/05core/usr/lib/systemd/system-generators/coreos-autologin-generator]: 👍

File: overlay.d/05core/usr/lib/systemd/system-generators/coreos-autologin-generator (hunk 4 above)
Comment (dustymabe) [reply overlay.d/05core/usr/lib/systemd/system-generators/coreos-autologin-generator]: 👍

File: overlay.d/05core/usr/lib/systemd/system-generators/coreos-autologin-generator (hunk 3 above)
Comment (dustymabe) [reply overlay.d/05core/usr/lib/systemd/system-generators/coreos-autologin-generator]: 👍

**Comment by dustymabe** [comment]
thanks @imcleod @jlebon for the review!

File: overlay.d/05core/usr/lib/systemd/system-generators/coreos-autologin-generator [hunk 5]
```diff
@@ -32,18 +32,41 @@ ExecStart=-/sbin/agetty --autologin core -o '-p -f core' ${args} %I \$TERM
...
 # nmtui and coreos-installer.
 kernel.printk=4
 EOF
 }
 
+write_interactive_live_issue() {
+    cat <<'EOF' > /etc/issue.d/80-interactive-live.issue
+
+###########################################################################
+Welcome to the Fedora CoreOS Live environment. This system is running
```
Comment (cgwalters) [inline overlay.d/05core/usr/lib/systemd/system-generators/coreos-autologin-generator:50]:
```
. /usr/lib/os-release
echo "Welcome to ${PRETTY_NAME} Live ...
```

File: overlay.d/05core/usr/lib/systemd/system-generators/coreos-autologin-generator [hunk 6]
```diff
@@ -32,18 +32,41 @@ ExecStart=-/sbin/agetty --autologin core -o '-p -f core' ${args} %I \$TERM
...
+completely from memory, making it a good candidate for hardware discovery
+and installing persistently to disk. Here is an example of running an
+install to disk via `coreos-installer`:
+
+curl -O https://example.com/example.ign
+sudo coreos-installer install /dev/sda \\
```
Comment (cgwalters) [inline overlay.d/05core/usr/lib/systemd/system-generators/coreos-autologin-generator:56]: I'm ok with this as is but I think we also should link to upstream docs directly.

File: overlay.d/05core/usr/lib/systemd/system-generators/coreos-autologin-generator (hunk 5 above)
Comment (dustymabe) [reply overlay.d/05core/usr/lib/systemd/system-generators/coreos-autologin-generator:50]: Sorry I didn't see this suggestion yesterday. Response over in https://github.com/coreos/fedora-coreos-config/pull/371#discussion_r417534510

File: overlay.d/05core/usr/lib/systemd/system-generators/coreos-autologin-generator (hunk 6 above)
Comment (dustymabe) [reply overlay.d/05core/usr/lib/systemd/system-generators/coreos-autologin-generator:56]: doc links would be nice

**Pull request merged**